larray/inout/csv.py

```diff
@@ -261,43 +261,47 @@ def __init__(self, fname, overwrite_file=False, sep=','):
         if fname is None:
             self.pattern = None
             self.directory = None
-        elif '.csv' in fname or '*' in fname or '?' in fname:
-            self.pattern = fname
-            self.directory = os.path.dirname(fname)
+        if not isinstance(fname, (str, Path)):
```

Comment:
this will break for fname is None! => use elif instead

also, you could simplify this slightly by moving the if isinstance(fname, str) check above